app/workers/find_and_delete_scheduled_accounts_worker.rb

```diff
@@ -4,6 +4,8 @@ class FindAndDeleteScheduledAccountsWorker
   include Sidekiq::Worker
 
   def perform
-    Account.deleted_since.find_each(&DeleteAccountHierarchyWorker.method(:perform_later))
+    Account.deleted_since.recently_changed_last.limit(100).find_each do |account|
```

Comment:
Imo it shouldn't be X limit a day but X limit at the same time. So when those X are finished, we should be able to go for another round of X.

About your question of this X being a config, it could be, but this does not belong to the account deletion feature (which is about automatically suspending and scheduling for deletion some accounts), this is about deleting accounts that have been scheduled for deletion at some point, no matter if manually or automatically. So if we make this 100 configurable, we should make also this other number configurable because they belong to the same feature:
https://github.com/3scale/porta/blob/40ea4fecfb8ad26f8ed4cbeb6c5dcde7b54af991/app/models/account/states.rb#L2

Reply:
you're right. we should re-run if there are more then X.

and if so, we don't have to introduce a new configuration for this number.. btw, when i gave it a second thought, it's shouldn't be configurable, it's way too "code" level.

Reply:
`find_in_batches`?

Reply:
@guicassolato as far as I know (and I could be wrong), `find_in_batches` executes the next batch even if the workers that it creates for the background didn't completely finish their job 🤔In that case, we have the same problem.

btw, `find_each` (what we are using) uses `find_in_batches` internally anyway, it is just that by default it is in batches of 1000, but we can send the number we want.

Reply:
`find_each(batch_size: 100)`

Reply:
That solves the 2nd paragraph of what I wrote 👍 but the 1st paragraph... I don't know, and it is important 😞

Reply:
@Martouta we could create a new batch, run workers within that batch, and delete another 100 accounts `on_complete` of that batch..

Reply:
Yeah, that first part is basically rate limiting. We went through a similar problem with billing and too much concurrency while setting invoices' friendly IDs a while ago. In the end we managed to delegate it to the db via stored procedures and simple sidekiq retries, but at some point we had considered a throttling mechanism such as https://github.com/sensortower/sidekiq-throttled.

Reply:
@macejmic , have in mind that if the batch callback fails, the whole chain is interrupted.

Reply:
that `sidekiq-throttled` looks super nice.. it would fix the problem right away. 

one long managed queue sounds much better then creating a new "managed" batch.

Reply:
@guicassolato it should be called even if the worker has failed (or one of them).. but true, better to not rely on that since we already had problem with those callbacks.